frontend/test/__support__/server-mocks/collection.ts

```diff
@@ -10,8 +10,11 @@ import {
 
 export function setupCollectionsEndpoints(
   scope: Scope,
-  collections: Collection[],
+  collections?: Collection[] | undefined,
 ) {
+  if (collections == null) {
+    collections = [ROOT_COLLECTION as unknown as Collection];
```

Comment:
This is not the correct way to do that.
1. `ROOT_COLLECTION` is not a `Collection`. It has some properties of the root collection but not all. To get a `Collection` you need to call `createMockCollection(ROOT_COLLECTION)`
2. Please do not add this implicit behavior of returning the root collection out of nowhever. Could you call `setupCollectionsEndpoints(scope, createMockCollection(ROOT_COLLECTION))` in your test?

Reply:
re: 1. it's fair to call createMockCollection(ROOT_COLLECTION), though it's worth pointing out that our API doesn't send the root collection with the same keys as other collections. The screenshot is taken when we're opening the save question modal.

<img width="827" alt="image" src="https://user-images.githubusercontent.com/1937582/213743238-0ec7b561-27b7-4dc4-b236-037f074e58f0.png">

You could see in this screenshot for fresh Metabase installation, there would be 2 collections, 1 for the root collection and the other one for admin collection.

re: 2. I still don't think this is implicit behavior, as explained above. Metabase will always have the root collection, so it's even more correct to return the root collection in `/api/collection` since we have mocked `/api/collection/root`. Think about calling `setupCollectionsEndpoints(scope)` without any parameter, what do you expect from this endpoint? If it's that we don't mock any `/api/collection`, but only mock `/api/collection/root` then isn't that weird, why would we deliberately mock just the collection instance, but not its list?

It seems like we mock `/api/collection/root` so that we don't have to care about the default value, but why would we expect people to pass the root collection along with other mock collections when we want to mock `/api/collection` (list)?

see this usage as an example.
https://github.com/metabase/metabase/blob/052ab0461d86505f24adc2f605e9835731e17da4/frontend/src/metabase/models/containers/ModelDetailPage/ModelDetailPage.unit.spec.tsx#L328-L331

those `collections` are then passed to:
https://github.com/metabase/metabase/blob/052ab0461d86505f24adc2f605e9835731e17da4/frontend/src/metabase/models/containers/ModelDetailPage/ModelDetailPage.unit.spec.tsx#L212

it's easy for engineers to only care about the collection they're testing. And it'd be a bit cumbersome to have every test have to pass the root collection there as well. In fact, it might be better to prepend the root collection to any passed `collection` (currently I only mock the root collection in the list endpoint if no `collections` is passed), so the way we mock collections endpoints is more consistent–if `/api/collection/root` exists then it should be present in `/api/collection` too.

tl;dr My comment might be hard to follow but I want to discuss why we'd choose to only mock `/api/collection/root`, but don't expect it to be in the response of `/api/collection/`, since in the actual Metabase instances, it's impossible to be able to call `/api/collection/:id` when that specific collection isn't returned in `/api/collection`.

Reply:
`Metabase will always have a root collection` - no, not always, depends on permissions

Reply:
I see your point that there is already an issue that there is this line:
```
scope.get("/api/collection/root").reply(200, ROOT_COLLECTION);
```

Not sure what to do about it. But the thing is that `ROOT_COLLECTION` is not always there fwiw.

Reply:
You're totally right. Haven't thought of that case. But still wouldn't you say what users can see should be consistent across the list and instance endpoint? I couldn't imagine a user couldn't see the root collection in `/api/collection` but can see `/api/collection/root`, can that happen?

On the other hand, I totally agree, we shouldn't have any default for `/api/collection` and I still appreciate mocking `/api/collection/root` for the sake of DX. It's worth leaving this as is for simplicity.

Reply:
Ok I think maybe there is a solution. We could add another parameter for the root collection:

```
setup...(
  scope, 
  collections,
  rootCollection = createMockCollection(ROOT_COLLECTION)
) {
 ...
 if (rootCollection.can_write) {
    scope.get("/api/collection/root").reply(200, rootCollection);
 } else {
    scope.get("/api/collection/root").reply(403);
 }
}
```

Reply:
@WiNloSt WDYT? I should fix the your case

Reply:
Oh, it will be wrong - it's ok to have it with `can_write` false :(((

Reply:
Ok, it's more complicated and I think we should change this helper in another PR. Let's not edit it here

Reply:
TBH, I don't know ins and outs of the collections yet. And would feel more comfortable leaving this as is for now. What I want to address is the consistency/data integrity, but seem it isn't that straightforward to fix. And that collection list fix doesn't have anything to do with this PR specifically, I only need to mock `/api/collection/root` for my tests.

Reply:
Reverted this file to `master` version.